test/CadlRanchProjects/type/property/value-types/src/Generated/Models/ExtendedEnum.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace _Type.Property.ValueTypes.Models
+{
+    /// <summary> The ExtendedEnum. </summary>
+    public readonly partial struct ExtendedEnum : IEquatable<ExtendedEnum>
```

Comment:
TCGC is better here. See the original cadl-ranch test: https://github.com/Azure/cadl-ranch/blob/eb9e34f6a00bf2df3f8a9ce2efdc286698cb8d8c/packages/cadl-ranch-specs/http/type/property/value-types/main.tsp#L278C60-L278C70
We should generate ExtentedEnum instead of UnionEnumValuePropertyProperty